Projects/hCore/Sources/ScrollView+Inspect.swift

```diff
@@ -33,17 +33,29 @@ struct ViewIntrospector<ViewType: UIView>: UIViewRepresentable {
         return nil
     }
 
+    func traverseUp(from view: UIView, levels: Int = 0) -> ViewType? {
+        if levels > 5 {
+            return nil
+        }
+
+        guard let superview = view.superview else {
+            return nil
+        }
+
+        if let view = findView(from: superview) {
+            return view
+        }
+
+        return traverseUp(from: superview, levels: levels + 1)
```

Comment:
Is there a reason to only go 5 levels up? Can't SwiftUI have a lot of stacked views?

Reply:
@tarikstafford the chance to actually find it that high up is low, and quite expensive, so to avoid traversing the whole hierarchy unnecessarily I thought it was good to cap it :)